src/store/compressors.rs

```diff
@@ -0,0 +1,155 @@
+use serde::{Deserialize, Serialize};
+use std::io;
+
+pub trait StoreCompressor {
+    fn compress(&self, uncompressed: &[u8], compressed: &mut Vec<u8>) -> io::Result<()>;
+    fn decompress(&self, compressed: &[u8], decompressed: &mut Vec<u8>) -> io::Result<()>;
+    fn get_compressor_id() -> u8;
+}
+
+/// Compressor can be used on `IndexSettings` to choose
+/// the compressor used to compress the doc store.
+///
+/// The default is Lz4Block, but also depends on the enabled feature flags.
+#[derive(Clone, Debug, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub enum Compressor {
+    /// Use the lz4 block format compressor
+    Lz4Block,
+    /// Use the lz4 frame format compressor
+    Lz4Frame,
```

Comment:
Is there a rationale behind having lz4frame AND lz4block?

If not could we just call it lz4 and pick the best one?

Reply:
I added the block format for backwards compatibility instead of replacing it. But since this is a breaking change anyway, I think we can just delete the lz4 frame stuff, since lz4 block is better imo in this case (less overhead and faster).

I think the removal with some ff renaming can be done in a separate PR